MailManager/src/main/java/inc/softserve/resources/EnvelopeResource.java

```diff
@@ -23,55 +24,27 @@
 @Produces(MediaType.APPLICATION_JSON)
 public class EnvelopeResource {
 
-    private EnvelopeDao envelopeDao;
-    private UserDao userDao;
-    private EnvelopeTools envelopeTools;
+    private EnvelopeService envelopeService;
 
     @Inject
-    public EnvelopeResource(EnvelopeDao envelopeDao, UserDao userDao, EnvelopeTools envelopeTools){
-        this.envelopeDao = envelopeDao;
-        this.userDao = userDao;
-        this.envelopeTools = envelopeTools;
+    public EnvelopeResource(EnvelopeService envelopeService) {
+        this.envelopeService = envelopeService;
     }
 
     @POST
     @Timed
     public Envelope saveEnvelope(@QueryParam("Subject") String subject,
                                @QueryParam("To") String to,
                                @QueryParam("From") String username,
-                               @QueryParam("template") String template,
-                               @Context HttpServletRequest request){
-        User from = userDao.getUserByUsername(username);
-
-        if(from == null) {
-            throw new IllegalArgumentException("No such user registered");
-        }
-
-        Envelope envelope = new Envelope(from, to, subject, template, EnvelopeState.IN_PROGRESS);
-        Envelope savedEnvelope = envelopeDao.saveOrUpdate(envelope);
-        EnvelopeState sendState = null;
-        try {
-            sendState = envelopeTools.sendEnvelope(savedEnvelope);
-        }catch(Exception e){
-            e.printStackTrace();
-            sendState = EnvelopeState.ERROR;
-        }
-
-        if(!EnvelopeState.IN_PROGRESS.equals(sendState)){
-            savedEnvelope.setState(sendState);
-            envelopeDao.saveOrUpdate(savedEnvelope);
-        }
-
+                               @QueryParam("template") String template){
+        Envelope savedEnvelope = envelopeService.saveOrUpdate(subject, to, username, template);

```

Comment:
I believe that we can immediately return the value, without using local variable.